File: 429-n-ary-tree-level-order-traversal/429-n-ary-tree-level-order-traversal.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */

/**
 * @param {Node|null} root
 * @return {number[][]}
 */
const levelOrder = function(root) {
    if (!root) return [];
    const q = [root];
    const levels = [];
    
    while (q.length) {
        const qLen = q.length; 
        const level = [];
        for (let i = 0; i < qLen; i++) {
            const node = q.shift();
            level.push(node.val);
            node.children.forEach(child => q.push(child));
        }
        levels.push(level);
    }
    
    return levels;
};

File: medium/429-n-ary-tree-level-order-traversal/429-n-ary-tree-level-order-traversal.rb
# Definition for a Node.
# class Node
#     attr_accessor :val, :children
#     def initialize(val)
#         @val = val
#         @children = []
#     end
# end

# @param {Node} root
# @return {List[List[int]]}
def level_order(root)
    res = []
    return res if !root
    
    q = Queue.new
    q << root
    
    while !q.empty? do 
        level = []
        
        q.size.times do 
            node = q.pop
            level << node.val
            node.children.each { |c| q << c }
        end
        
        res << level
    end
    
    res
end